pkg/git/v2/executor.go

```diff
@@ -63,7 +79,14 @@ type censoringExecutor struct {
 }
 
 func (e *censoringExecutor) Run(args ...string) ([]byte, error) {
-	logger := e.logger.WithField("args", strings.Join(args, " "))
+	censoredArgs := make([]string, len(args))
+	for i, arg := range args {
+		censoredArgs[i] = censorURLCredentials(arg)
```

Comment:
I wonder if  the existing censoring method (which has an advantage of censoring all secrets possibly known to the process) could be enough here, like:

```suggestion
		censoredArgs[i] = e.censor(arg)
```

(plus the bytes/string conversion)

 Have you tried that?

But the _actual_ benefit of that is likely just hypothetical compared to your fix (_other_ secrets leaked through _other_ arguments, hypothetical) so /shrug